### DISCUSSION TITLE
v0.6.3-rc3 Release Candidate

### DISCUSSION THREAD

**Discussion by fvanroie** [HASwitchPlate/openHASP#334] [Releases]
v0.6.3 has been in development for quite some time. It has a long [CHANGELOG ](https://github.com/HASwitchPlate/openHASP/blob/master/CHANGLELOG.md) with a lot of bug fixes and additional features.

I think this update warrants a more thorough testing process before freezing the code and doing an official release... Hence I would like to ask you to help testing.

**UPDATE:** I'm currently testing [v0.6.3-rc2](https://github.com/HASwitchPlate/openHASP/actions/runs/2371784404) 1c86a3a with these hardware devices:
- [x] az-touch mod
- [x] esp32-d1-mini
- [x] esp32-touchdown
- [x] freetouchdeck_8MB
- [x] m5stack-core2
- [x] wt32-sc01_4MB

other testers:
- [x] lanbon-l8_8MB (@nagyrobi)

You can download or install [v0.6.3-rc2](https://github.com/HASwitchPlate/openHASP/actions/runs/2371784404) 1c86a3a from https://install.openhasp.com

If you are testing v0.6.3-rc3 on another device, please report below.
You can also provide all feedback in this thread (or the Github issue if it already exists).

I'm going to update the [v0.6.3 documentation](https://openhasp.haswitchplate.com/0.6.3/) next, so stay tuned for that.

**Comment by nagyrobi** [comment]
- Lanbon L8 updated smoothly on 5 production devices from a november 0.6.3-dev (af336e4ec1163a0fc699cd1bba06517b94b99e29) to a fresh v0.6.3-rc1 (5e333e8)
- memory usage and fragmentation gets higher with core v.4.4.1 than with v.4.4.0.
- TTF support works great

**Reply by nagyrobi** [reply]
Plethora of font sizes and TTF-based icons, PNG images:
![image](https://user-images.githubusercontent.com/1550668/167477461-9496ed9b-64c6-4467-95cf-f2b5cf8c575d.png)  ![image](https://user-images.githubusercontent.com/1550668/167477515-8ed8b4ab-9729-459f-895a-b4c4c940de67.png)  ![image](https://user-images.githubusercontent.com/1550668/167477593-42420cb2-e893-4442-b73e-6b41b5e1ee06.png)  ![image](https://user-images.githubusercontent.com/1550668/167477930-adf4aabe-ec19-435b-9109-38abaf07f212.png)  ![image](https://user-images.githubusercontent.com/1550668/167478021-d9cce3c0-013b-4703-847e-b61d29263252.png)  ![image](https://user-images.githubusercontent.com/1550668/167478511-8acfd753-5f9a-4b4b-8df9-3b4b475297f3.png)

I don't think we ever need anything more than this!

**Comment by mvturnho** [comment]
Did some testing on WT32-SC01 and the sleep mode does not work. 
I see the events getting send over mqtt. But the backlight remains lit, and the device is not sleeping.

**[13:07:38.624][79860/83588  4][55248/55788  1] MQTT PUB: idle => short
[13:09:38.617][79860/83808  4][55772/56356  2] MQTT PUB: idle => long**

**Reply by nagyrobi** [reply]
Do you use the Custom Component with HA?

**Reply by mvturnho** [reply]
Ok got this working, this is configured in the mqtt integration with homeassistant.

**Reply by mvturnho** [reply]
> 

thanks for the hint ;-)

**Reply by fvanroie** [reply]
Thanks for testing and reporting back

**Comment by mmccool** [comment]
I am planning to test this release candidate on the Makerfab ESPTFT35CA and integrating with Home Assistant.  I am compiling myself using VSCode/Platform IO under Ubuntu, with a custom configuration (just to set up WiFi and MQTT), and will be testing both serial install and OTA.  [See related issue on enabling the backlight on this board.](https://github.com/HASwitchPlate/openHASP/issues/335)

Status Updates:
- Cloned via git to Linux system
- WiFi and MQTT configured by editing templates in code
- Compiled using code/Platform IO
- 0.6.3-rc1 loaded and running
- Integrated with Home Assistant using HACL Integration
- Serial port disabled, GPIO1 enabled as PWM using Web UI
    - Haven't yet connected GPIO1 to the backlight control, will do soon, but probably only after I play around with configurations etc. so as to not complicate testing the release candidate
    - Despite enabling GPIO1 as a PWM and rebooting both the board and Home Assistant, GPIO1 is not showing up as a controllable entity in Home Assistant.  The configuration one the plate, as seen via the Web UI, is persistent across reboots.  It may be I can configure it manually, but it seems automatic entity detection is not working.  Note that I did the GPIO configuration after I had already installed and set up the HA integration.
- After [reading](https://openhasp.haswitchplate.com/0.6/integrations/home-assistant/howto/) that versions of openHASP integration in Home Assistant and the #version of firmware on the plates must match, I checked the version of the HACS integration I had installed and it was 0.6.5.  The "5" is confusing.  If 0.6.3-rc1 is the latest version, how can there be a 0.6.5 integration?  Should be be rolling the integration back to 0.6.3, or does 0.6.5 work with earlier versions of the firmware?  Anyway, I figured out HOW to roll it back if I have to, but will leave the integration at 0.6.5 for now.
- Tried to set up Home Assistant script to upload pages.json and/or pages.jsonl from /config/openhasp/pages.json(l) file under HA but haven't been able to get it to work.  Here is the script:
```
alias: Reload openHASP Page Configuration
sequence:
  - service: openhasp.load_pages
    data:
      path: /config/openhasp/pages.jsonl
    target:
      entity_id: openhasp.plate01
mode: single
icon: mdi:reload
```
and the following is in my configuration.yaml as well:
```
homeassistant:
    ... (other stuff)
    allowlist_external_dirs:
      - "/config/openhasp"
```
the script executes with no errors but the configuration does not seem to "take".   I tried both json and jsonl.  The jsonl configuration file works if I upload it via the Web UI (which I figured out works only for uploading but not for editing...) but not the .json even if I edit the config also to point at "pages.json".  Finally, the descriptions of the openhasp.load_pages service still says jsonl but does not mention json.
- Tried rolling the HACS integration back to version 0.6.3 and now the upload script above works.  Alrighty then.
- Upload works in the sense that the display on the device updates, but when I try to check the file using the WebUI I only see the old one.  Not what I was expecting, I would have thought the Web UI would let me see the *current* status of the configuration files.
- Confirmed both pages.jsonl and pages.json work.  Proceeding with pages.json although it makes using the examples more work; I have to convert them to json first.  Going to now try to use the "Weather" example, but add a page that shows values from my Weatherflow Tempest system as well.


Comments:
- I'm new to openHASP and so am setting up my configuration for the first time.  However, I have a Home Assistant setup with about 700 entities I am trying to integrate with.
- Using the web UI to try and update the pages.jsonl file is very frustrating, as there is no obvious way to "save" edits.  I want to do it another way anyway, and this may be a general issue and not a 0.6.3-rc1 issue, but it would be nice if this part of the web UI was improved.  So far I can only get the "upload" functionality to work.  It seems editing should work but doesn't.  I am using Chrome on a Mac.
- I tried setting up a template string with two values in it, and one had an error, but then neither rendered, or rather, the system now just displays the uninterpreted template string.  The string I tried to use was "%hostname% %id%" (I actually wanted to render "%hostname% %ip%" but mistyped)
- The Makerfab board I am using compiled without errors, but some of the other boards I tried had compile errors.  Stale/untested configurations?

**Reply by nagyrobi** [reply]
> there is no obvious way to "save" edits

CTRL+S (on PC at least), I make edits on the web interface, and reload design there to see changes.

I like to keep the jsonl files on the plate, not in HA, and that's because [batch processing](https://openhasp.haswitchplate.com/0.6.3/commands/#batch-processing) allows loading various page sets even when certain connection conditions are not met (like when MQTT is down).

**Reply by nagyrobi** [reply]
> I had installed and it was 0.6.5.  The "5" is confusing.

"5" is minor version. In the docs it's "x" so any 0.6.x with any other 0.6.x should match well.

**Reply by fvanroie** [reply]
"5" actually is the revision, so it's major.minor.revision. As long as major.minor versions match it should be okay. The revision isn't taken into account. You most definitely want the `v0.6.5` branch of the _Custom Component_ or even `main` branch.

> some of the other boards I tried had compile errors. Stale/untested configurations?

Well it depends on which boards you enabled in the `platformio_override.ini`. Not all configurations are "officially" supported or actively maintained. There are user-contributed ones, test devices, new devices... You shouldn't do `Build all` unless you cherry-pick the boards first.

>  it would be nice if this part of the web UI was improved

The File Browser is heavily borrowed from [Espressifs FSBrowser example](https://github.com/espressif/arduino-esp32/tree/master/libraries/WebServer/examples/FSBrowser) with some minor tweaks to make it usable. I don't have the knowledge to reverse engineer or rewrite this part, so if anyone who's fluent in Javascript wants to take a stab at this I'll happily accommodate.

**Reply by fvanroie** [reply]
> The string I tried to use was "%hostname% %ip%"

There is no *real* template support, you can only use one "template" at the time... If you want to show both the hosname and ip, create two labels instead.

**Reply by mmccool** [reply]
- Well, regarding the integration, 0.6.5 did not work, but 0.6.3 did.  So while it may be *supposed* to work... it doesn't.  Also, the docs say "exactly match".  Perhaps they should say "major and minor versions should match".
- Regarding the boards, some indication of which are "supported" and which are not would be helpful. 
- "No obvious way to save": sorry, Ctrl-S is not obvious.  I generally use vi on Linux, not Windows, etc. (I'm also a CLI/Makefile type not an IDE person but that's a different discussion...).  Anyway, even a label somewhere saying "Ctrl-S to save" or a note in the docs would have been helpful.
- Regarding keeping the pages.jsonl on HA or on the device: HA is backed up.  Devices can die and then what?  Generally after doing all the work of setting up a configuration I want it in a safe place.
- Yeah, I figured multiple templates would not work.  It would be nice if they did, though, since it would make formatting easier, i.e. then you don't have to worry about changes in label sizes and placing labels so they don't clobber each other, etc.  So consider it a feature request.  Anyway, I mention it only because it triggered another problem I ran into.

**Reply by fvanroie** [reply]
Regarding the boards: https://github.com/HASwitchPlate/openHASP/blob/master/.github/workflows/build.yaml#L16-L50
And the File Browser: https://openhasp.haswitchplate.com/0.6.3/faq/#how-to-save-changes-in-the-file-editor-to-flash
Ctrl-S is not obvious, I don't disagree there but that's just how it is at the moment...

I don't think the other items are related to RC1. You can report issues with the [docs](https://github.com/HASwitchPlate/openHASP-docs) or [HA custom component](https://github.com/HASwitchPlate/openHASP-custom-component) in their respective repos.
Multiple people maintain different parts, so it will be the quickest to open an *issue* or *pull request* there.

**Comment by thirug010** [comment]
V0.6.3 rc1 looks great thanks for updating it, but I observed some issue with the Antiburnin. it is not turning off automatically or with the mqtt with off message or when the screen is touched. 

it only turns off when page is navigated through navigation button is pressed.

Tested with ota and full firmware update on 
1. Lanbon L8 
2. wt32-sc01 (16MB)

**Reply by nagyrobi** [reply]
> I observed some issue with the Antiburnin.

Fixed in RC2

**Reply by thirug010** [reply]
Thank you, can you provide me the link for rc2 .. ?

**Reply by nagyrobi** [reply]
Soon...

**Reply by fvanroie** [reply]
You can download it from https://install.openhasp.com/ or if you have Chrome or Edge it also can be installed from within the web browser.
I'm still testing the web installer, so use with care and maybe make a backup first just to be safe.

**Reply by thirug010** [reply]
Anti burn issue is fixed RC2 - Thanks

**Comment by id10tapproved** [comment]
RC1 is working well on a custom build: LOLIN D32 Pro with ILI9488 Display and GT911 Capacitive Touch Interface. 

Seems like it boots much faster than 0.6.2.

**Reply by fvanroie** [reply]
Thanks, that's good to know. Could you share the display link or model?

**Reply by thirug010** [reply]
@fvanroie I found one with ILI9488 Display and GT911 Capacitive Touch Interface. 
https://www.aliexpress.com/item/32356556468.html?spm=a2g0o.cart.0.0.32923c00l1MkQQ&mp=1

**Reply by id10tapproved** [reply]
@fvanroie This is the model I am using. (it's cheaper than the one @thirug010 linked to) 
https://www.aliexpress.com/item/1005004083478309.html?spm=a2g0o.order_list.0.0.3b4d1802Zo4VHB

**Reply by fvanroie** [reply]
Thanks those both seem very nice displays. One is parallel and the other is serial which can explain some of the price difference.

**Comment by fvanroie** [comment]
The latest 0.6.3-rc2 is now available for testing from https://install.openhasp.com/ which fixes the latest known bugs from rc1.

**Comment by fvanroie** [comment]
We have found a bug regarding `value_str` on parts and states. This has now been fixed in 0.6.3-rc3.
This should be the last before release 🤞

**Comment by devinmitchell** [comment]
Is there any possible way to have a command run, or a page loaded when connection to MQTT is down?  I have 10 Lanbon L8s in the house and whenever I reboot HA all our light switches just show "Connecting..".  This normally isn't a problem, but recently an HA update failed which resulting in me needing to roll back to a snapshot and restore from a backup, which resulted unusable switches for a while.
I have the following relevant files configured as such:
`offline.cmd`:
```
clearpage all
moodlight {"state":true,"color":"#FFFF00"}
run /pages_offline.jsonl
```

`online.cmd`:
```
service stop console
jsonl {"page":0,"id":239,"obj":"msgbox","txt":"%ip%", "auto_close":5000}
clearpage all
run /pages_online.jsonl
```

`pages_offline.cmd`:
```
{"page":1,"id":4,"obj":"label","x":0,"y":5,"h":30,"w":240,"text":"Offline...","text_color":"white","align":1,"bg_color":"#2C3E50"}

{"page":1,"id":1,"obj":"btn","x":10,"y":40,"w":220,"h":60,"toggle":true,"text":"\uE335 Outside","text_font":32,"align":1,"groupid":1}
{"page":1,"id":2,"obj":"btn","x":10,"y":120,"w":220,"h":60,"toggle":true,"text":"\uE335 Spots","text_font":32,"align":1,"groupid":2}

{"page":0,"id":11,"obj":"btn","action":"prev","x":0,"y":290,"w":79,"h":32,"bg_color":"#2C3E50","text":"\uE141","text_color":"#FFFFFF","radius":0,"border_side":0,"text_font":28}
{"page":0,"id":12,"obj":"btn","action":"back","x":80,"y":290,"w":80,"h":32,"bg_color":"#2C3E50","text":"\uE2DC","text_color":"#FFFFFF","radius":0,"border_side":0,"text_font":22}
{"page":0,"id":13,"obj":"btn","action":"next","x":161,"y":290,"w":79,"h":32,"bg_color":"#2C3E50","text":"\uE142","text_color":"#FFFFFF","radius":0,"border_side":0,"text_font":28}
{"page":0,"id":9,"obj":"obj","swipe":1,"x":0,"y":0,"h":50,"w":240,"opacity":0,"comment":"swipe-area-at-top"}
```

...and `pages_online.jsonl` is configured with the normal display which I won't post here as it's too long and not relevant.

If I unplug the WiFi, pages_offline.jsonl is displayed which allows me to turn on/off the relays which is great, but I'd give my left toe to gave a page load if the device is unable to communicate with MQTT.  :)

**Reply by fvanroie** [reply]
Check out #61 and #174 . It's not part of this release, but will be included in a future release.
Please comment in the linked topics so they can be taken into account when the feature is implemented.

**Comment by fvanroie** [comment]
Thanks everyone for testing.  v0.6.3 has now been released #338 !
